Source/DiabloUI/diabloui.cpp

```diff
@@ -672,6 +672,11 @@ Sint16 GetCenterOffset(Sint16 w, Sint16 bw)
 	return (bw - w) / 2;
 }
 
+void UiLoadBlackBackground()
+{
+	LoadBackgroundArt(gbIsHellfire ? "ui_art\\black_hellfire.pcx" : "ui_art\\black_diablo.pcx");

```

Comment:
Wouldn't this work just as well?
```suggestion
	LoadBackgroundArt(gbIsHellfire ? "ui_art\\black.pcx" : "ui_art\\black_diablo.pcx");
```
That way we don't need an embeded file for Hellfire.

Reply:
Our embedded file is much smaller than what's shipped with Hellfire, at only 899 bytes vs 11.5 KiB, so that's a nice advantage of shipping our own as well

Reply:
Speaking of background issues in the settings men :thinking: 

![image](https://user-images.githubusercontent.com/204594/176791867-48a2bad5-73a3-457a-a856-b0ff6d1123f9.png)

Reply:
> Our embedded file is much smaller than what's shipped with Hellfire, at only 899 bytes vs 11.5 KiB, so that's a nice advantage of shipping our own as well

Is it worth it, since it will be unloaded before entering the game?

Reply:
1. Loading `ui_art/black.pcx` won't work well if `gbIsHellfire` is true and there is no `hellfire.mpq`.

2. The UI glitch always happens at 640x480 (because of how `UiClearScreen()` works together with our 1px black background).

I'll fix the glitch and clean up the black background stuff a bit more.

Memory thrashing before starting a game can still impact the game on super memory-constrained devices because of malloc fragmentation etc.

Reply:
Done